File: 3872-find-most-frequent-vowel-and-consonant/find-most-frequent-vowel-and-consonant.cpp
class Solution {
public:
bool isvowel(char c){
    if(c=='a' || c=='e' || c=='i' || c=='o' || c=='u') return true;
    return false;
}
    int maxFreqSum(string s) {
    unordered_map<char,int> mp;
    for(int ele:s){
        mp[ele]++;
    }  
    int maxvowel=0;
    int maxcons=0;

    for(auto el:mp){
        if(isvowel(el.first)) maxvowel=max(maxvowel,el.second);
        else maxcons=max(maxcons,el.second);
    }
    return maxcons+maxvowel;
    }
};

File: 3872-find-most-frequent-vowel-and-consonant/find-most-frequent-vowel-and-consonant.java
class Solution {
    public int maxFreqSum(String s) {
        int[] freq = new int[26];
        for (char ch : s.toCharArray()) {
            freq[ch - 'a']++;
        }

        int maxVowelFreq = 0, maxConsonantFreq = 0;
        String vowels = "aeiou";

        for (int i = 0; i < 26; i++) {
            char c = (char) (i + 'a');
            if (freq[i] > 0) {
                if (vowels.indexOf(c) != -1) { 
                    // It's a vowel
                    maxVowelFreq = Math.max(maxVowelFreq, freq[i]);
                } else {
                    // It's a consonant
                    maxConsonantFreq = Math.max(maxConsonantFreq, freq[i]);
                }
            }
        }

        return maxVowelFreq + maxConsonantFreq;
    }
}
